src/transactions/swap.js

```diff
@@ -154,77 +165,25 @@ export async function swapAlgoToGard(amount, minimum) {
         to: info.address,
         amount: 0,
         suggestedParams: params,
-        assetIndex: gardID,
+        assetIndex: toAsset.index,
       });
-  let txns = optTxn.concat([txn1, txn2]);
-  algosdk.assignGroupID(txns);
-
-  setLoadingStage("Awaiting Signature from Algorand Wallet...");
-
-  const signedGroup = await signGroup(info, txns);
-
-  setLoadingStage("Waiting for Confirmation...");
 
-  const stxns = [signedGroup[0].blob, signedGroup[1].blob];
-
-  const response = await sendTxn(
-    stxns,
-    "Successfully swapped " + amount / 1e6 + " ALGO.",
-  );
-
-  setLoadingStage(null);
-  updateDBWebActions(8, null, -amount, minimum, 0, 1, 3000);
-  return response;
-}
-
-/**
- * @function swapGardToAlgo - create and send transaction to swap GARD FOR ALGO on Pact DEX
- *    @param {amount} {number} - representing how many microgard to send
- *    @param {minimum} {number} - representing minimum acceptable microAlgos for swap to succeed
- *    @returns {transactionSummary} - returns details of the exchange to allow for execution
- */
-export async function swapGardToAlgo(amount, minimum) {
-  setLoadingStage("Loading...");
-
-  const infoPromise = accountInfo();
-  const paramsPromise = getParams(1500);
-  const info = await infoPromise;
-  const params = await paramsPromise;
-  const f_a = [0, gardID];
-  const enc = new TextEncoder();
-
-  let txn1 = algosdk.makeAssetTransferTxnWithSuggestedParamsFromObject({
-    from: info.address,
-    to: pactAlgoGardPoolAddress,
-    amount: amount,
-    suggestedParams: params,
-    assetIndex: gardID,
-  });
-  let txn2 = algosdk.makeApplicationCallTxnFromObject({
-    from: info.address,
-    appIndex: pactGARDID,
-    onComplete: 0,
-    appArgs: [enc.encode("SWAP"), algosdk.encodeUint64(minimum)],
-    foreignAssets: f_a,
-    suggestedParams: params,
-  });
-
-  let txns = [txn1, txn2];
+  let txns = Array.isArray(optTxn) ? optTxn.concat([txn1, txn2]) : [txn1, txn2];
   algosdk.assignGroupID(txns);
 
   setLoadingStage("Awaiting Signature from Algorand Wallet...");
   const signedGroup = await signGroup(info, txns);
-
   setLoadingStage("Waiting for Confirmation...");
 
   const stxns = [signedGroup[0].blob, signedGroup[1].blob];
-
   const response = await sendTxn(
     stxns,
-    "Successfully swapped " + amount / 1e6 + " GARD.",
+    "Successfully swapped " +
+      formattedAmount / 1e6 +
```

Comment:
We should fetch the decimals of each asset instead of hard-coding 6 here. I suspect we already have a asset info call somewhere.